test/unittests/mongoBackend/mongoContextProvidersQuery_test.cpp

```diff
@@ -541,8 +541,9 @@ static void prepareDatabaseSeveralCprs2(void)
 *
 * noPatternAttrsAll -
 *
-* Discover:  E3 -no attrs
-* Result:    E3 - (A1, A2, A3) - http://cr1.com
+* Query:  E3 - <null>
+* Result: E3 - (A1, A2, A3) - http://cr1.com
+*
 */
 TEST(mongoContextProvidersQueryRequest, noPatternAttrsAll)
```

Comment:
Appart of the three new tests (severalCprsX) the old test has been modified and adapted. Now, they are not based on 302 Found but they work in a more proper way.

I would be great if you could have a look and check that everything works as you will expectd in the postQueryContext serviceRoutine (as they will play the role than the unit tests is taking now but for real).